gap/attributes/cartan.gi

```diff
@@ -0,0 +1,741 @@
+
+InstallMethod(RClassRadicalBicharacterOfGroupHClass,  "for group H-class",
+[IsGroupHClass],
+function(H)
+  local S, e, Rec, Rad, LHH, map, invmap, r_mults, rp_mults,
+        ListLClass, n, HH, CHH, ord, B, dim,
+        CS, cS, CG, cG, mat, compt,
+        h, k, chi, ind_r, r, row, i, coeff,
+        ind_transition, ind_groupe, x, lp, g, ind_l_class;
+
+  S   := ParentAttr(H);
+  e   := MultiplicativeNeutralElement(H);
+
+  CS   := GeneralizedConjugacyClassesRepresentatives(S);
+  cS   := Length(CS);
+
+  map := IsomorphismPermGroup(H);
+  invmap := InverseGeneralMapping(map);
+
+  HH  := Range(map);
+  CHH  := ConjugacyClasses(OrdinaryCharacterTable(HH));
+
+  CG   := List(List(CHH, Representative), x -> x ^ invmap);
+  cG   := Length(CG);
+
+  Rec := RClassRadicalOfGroupHClass(H);
+  Rad := Rec.rad;
+  LHH := Rec.HList;
+  r_mults  := Rec.transitions;
+  rp_mults := Rec.returns;
+
+  ListLClass := List(r_mults, r -> LClass(S, e * r));
+
+  if Length(Rad) = 0 then
+    Rad := [[0]];
+  fi;
+  n    := Length(Rad[1]);
+  ord  := Length(LHH);
+  B    := Basis(VectorSpace(Rationals, Rad));
+  dim  := Length(B);
+
+  mat  := List([1 .. cG], x -> List([1 .. cS], x -> 0));
+
+  for h in CS do
+    for k in CG do
+      chi := 0;
+
+      # Computing the contribution to the trace of each basis vector
+      for ind_r in [1 .. dim] do
+        r   := B[ind_r];
+        row := List([1 .. n], x -> 0);
+        compt := 0;
+        # Computing the image of the vector
+        for i in [1 .. n] do
+          coeff := r[i];
+          if coeff = 0 then continue;
+          fi;
+          ind_transition := QuoInt(i - 1, ord) + 1;
+          ind_groupe := RemInt(i - 1, ord) + 1;
+          x := k * LHH[ind_groupe] * r_mults[ind_transition] * h;
+          ind_transition := Position(ListLClass, LClass(S, x));
+          # Changed from not ind_transition = fail
+          if ind_transition <> fail then
+            compt := compt + 1;
+
+            lp := rp_mults[ind_transition];
+
+            g  := (e * x * lp) ^ map;
+            ind_groupe  := Position(LHH, g);
+            ind_l_class := (ind_transition - 1) * ord + ind_groupe;
+            row[ind_l_class] := row[ind_l_class] + coeff;
+          fi;
+        od;
+        chi := chi + Coefficients(B, row)[ind_r];
+      od;
+      mat[Position(CG, k)][Position(CS, h)] := chi;
+    od;
+  od;
+
+  SetRClassRadicalBicharacterOfGroupHClass(H, mat);
+
+  return mat;
+end);
+
+InstallMethod(DiagonalOfCharacterTables,  "for a semigroup",
+[IsSemigroup],
+function(S)
+  # Removed loval variable CS
+  local M, transversalHclasses, maps, groups, charactertables,
+      irrs, mats;
+
+  # Removed following line of code as a part of linting.
+  # The following line of code was run early to ensures that the
+  # conjugacy classes were computed in the same order each time.
+  # As I have leared more about the GAP language this step might be
+  # unnesssisary. Until I am sure, I will leave this line here with
```

Comment:
```suggestion
  # unnecessary. Until I am sure, I will leave this line here with
```

This doesn't really impact the PR, but spelling mistakes might be caught by code spell in the future, and so better to minimise them.